packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
+  /// [update].
   final Uri basedir;
 
   /// The file system used to perform file access.
   @visibleForTesting
   final FileSystem fs;
 
-  /// Creates a new [FlutterGoldenFileComparator] that mirrors the relative
-  /// path resolution of the default [goldenFileComparator].
+  /// A wrapper for the [dart:io.Platform] API.
+  ///
+  /// This is only for use in tests, where the system platform (the default) can
+  /// be replaced by a mock platform instance.
+  @visibleForTesting
+  final Platform platform;
+
+  /// Creates a new [FlutterGoldensRespositoryFileComparator] that mirrors the
+  /// relative path resolution of the default [goldenFileComparator].
   ///
   /// By the time the future completes, the clone of the `flutter/goldens`
   /// repository is guaranteed to be ready use.
   ///
   /// The [goldens] and [defaultComparator] parameters are visible for testing
   /// purposes only.
-  static Future<FlutterGoldenFileComparator> fromDefaultComparator({
-    GoldensClient goldens,
+  static Future<FlutterGoldensRepositoryFileComparator> fromDefaultComparator({
+    GoldensRepositoryClient goldens,
     LocalFileComparator defaultComparator,
   }) async {
     defaultComparator ??= goldenFileComparator;
 
     // Prepare the goldens repo.
-    goldens ??= GoldensClient();
+    goldens ??= GoldensRepositoryClient();
     await goldens.prepare();
 
     // Calculate the appropriate basedir for the current test context.
     final FileSystem fs = goldens.fs;
     final Directory testDirectory = fs.directory(defaultComparator.basedir);
     final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
-    return FlutterGoldenFileComparator(goldens.repositoryRoot.childDirectory(testDirectoryRelativePath).uri);
+    return FlutterGoldensRepositoryFileComparator(goldens.comparisonRoot.childDirectory(testDirectoryRelativePath).uri);
   }
 
+  /// Compares [imageBytes] against the golden file identified by [golden].
+  ///
+  /// The returned future completes with a boolean value that indicates whether
+  /// [imageBytes] matches the golden file's bytes.
+  ///
+  /// In the case of the given golden file being unavailable, the comparator
+  /// will throw a [TestFailure].
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
-    final File goldenFile = _getGoldenFile(golden);
+    final File goldenFile = getGoldenFile(golden);
     if (!goldenFile.existsSync()) {
       throw TestFailure('Could not be compared against non-existent file: "$golden"');
     }
-    final List<int> goldenBytes = await goldenFile.readAsBytes();
-    // TODO(tvolkert): Improve the intelligence of this comparison.
-    if (goldenBytes.length != imageBytes.length) {
-      return false;
-    }
-    for (int i = 0; i < goldenBytes.length; i++) {
-      if (goldenBytes[i] != imageBytes[i]) {
+
+    if (platform.isLinux) {
```

Comment:
Instead of doing this, we could also have a third dummy `FlutterGoldenFileComparator` impl that always skips and in main you pick that one if we can't do skia gold and are not on Linux. (It is kinda surprising that FlutterGoldensRepositoryFileComparator wouldn't work on Linux)